### PR TITLE
chore(external docs): Rename auto concurrency to adaptive concurrency

### PR DESCRIPTION
**Pull request by lucperkins**
This PR updates both the CUE documentation sources and the Rust codebase to use "adaptive" rather than "auto" concurrency. This includes changes to HTTP sink configuration:

```toml
# before
request.in_flight_limit = "auto"

# now
request.concurrency = "adaptive"
```

The `concurrency` parameter has been aliased to allow for `in_flight_limit` to preserve backward compatibility.

@bruceg I added just one simple test (`backward_compatibility_with_in_flight_limit_param_works`) for that. Please let me know if this is insufficient and I'd be happy to make that more robust.

### CODE REVIEW

**Review by bruceg — APPROVED** [review]
Looks fine to me, and the one test for `in_flight_limit` compatibility should be adequate, at least until we discover other issues with it.

File: tests/data/auto-concurrency-template.toml [hunk 1]
```diff
@@ -17,7 +17,7 @@ rate.knee_exp = 0.0
 rate.limit = 0
 rate.action = "defer" | "drop"
 
-in_flight_limit = "auto"
+concurrency = "auto"
```
[the remark this answers is not in the document]
Comment (lucperkins) [reply tests/data/auto-concurrency-template.toml:20]: Good catch

File: docs/reference/components/sinks.cue [hunk 2]
```diff
@@ -203,7 +203,7 @@ components: sinks: [Name=string]: {
 					type: object: {
 						examples: []
 						options: {
-							auto_concurrency: {
+							adaptive_concurrency: {
 								common:      false
 								description: "Configure the auto-concurrency algorithms. These values have been tuned by optimizing simulated results. In general you should not need to adjust these."
```
[the remark this answers is not in the document]
Comment (lucperkins) [reply docs/reference/components/sinks.cue]: Good catch

**Comment by binarylogic** [comment]
@lucperkins let's get this in before the release since it would be a breaking change otherwise. And it aligns with our marketing.

**Comment by lucperkins** [comment]
@binarylogic I'll do what I can but I'm a bit stymied by the CI issues

**Comment by binarylogic** [comment]
You can ignore the " Publish Docs If Updated / build (pull_request)" failure, the others should pass though.

**Comment by binarylogic** [comment]
@Hoverbear @MOZGIII @lucperkins can we try to wrap this up on Friday so we can get it in the release? @MOZGIII @Hoverbear pinging you to help with any Rust issues and to review the changes.

**Review by Hoverbear — APPROVED** [review]

**Pull request merged**